src/Tmds.Ssh/KerberosCredential.cs

```diff
@@ -0,0 +1,294 @@
+                out statusCode);
+        }
+
+        return statusCode == NegotiateAuthenticationStatusCode.Completed;
+    }
+
+    private static Packet CreateOidRequestMessage(SequencePool sequencePool, string userName, ReadOnlySpan<byte> oid)
+    {
+        /*
+            byte      SSH_MSG_USERAUTH_REQUEST
+            string    user name (in ISO-10646 UTF-8 encoding)
+            string    service name (in US-ASCII)
+            string    "gssapi-with-mic" (US-ASCII method name)
+            uint32    n, the number of mechanism OIDs client supports
+            string[n] mechanism OIDs
+        */
+        using var packet = sequencePool.RentPacket();
+        var writer = packet.GetWriter();
+        writer.WriteMessageId(MessageId.SSH_MSG_USERAUTH_REQUEST);
+        writer.WriteString(userName);
+        writer.WriteString("ssh-connection");
+        writer.WriteString(AUTH_METHOD);
+        writer.WriteUInt32(1);
+        writer.WriteString(oid);
+        return packet.Move();
+    }
+
+    private static Packet CreateGssapiTokenMessage(SequencePool sequencePool, ReadOnlySpan<byte> token)
+    {
+        /*
+            byte        SSH_MSG_USERAUTH_GSSAPI_TOKEN
+            string      data returned from either GSS_Init_sec_context()
+                        or GSS_Accept_sec_context()
+        */
+        using var packet = sequencePool.RentPacket();
+        var writer = packet.GetWriter();
+        writer.WriteMessageId(MessageId.SSH_MSG_USERAUTH_GSSAPI_TOKEN);
+        writer.WriteString(token);
+        return packet.Move();
+    }
+
+    private static Packet CreateGssapiCompleteMessage(SequencePool sequencePool)
+    {
+        /*
+            byte      SSH_MSG_USERAUTH_GSSAPI_EXCHANGE_COMPLETE
+        */
+        using var packet = sequencePool.RentPacket();
+        var writer = packet.GetWriter();
+        writer.WriteMessageId(MessageId.SSH_MSG_USERAUTH_GSSAPI_EXCHANGE_COMPLETE);
+        return packet.Move();
+    }
+
+    private Packet CreateGssapiMicData(SequencePool sequencePool, ReadOnlySpan<byte> sessionId, string userName, NegotiateAuthentication authContext)
+    {
+        /*
+            string    session identifier
+            byte      SSH_MSG_USERAUTH_REQUEST
+            string    user name
+            string    service
+            string    "gssapi-with-mic"
+        */
+        using var packet = sequencePool.RentPacket();
+        var writer = packet.GetWriter();
+        writer.WriteString(sessionId);
+        writer.WriteMessageId(MessageId.SSH_MSG_USERAUTH_REQUEST);
+        writer.WriteString(userName);
+        writer.WriteString("ssh-connection");
+        writer.WriteString(AUTH_METHOD);
+
+        // The MIC data does not include the header, so skip the first 5 bytes.
+        ReadOnlySequence<byte> micData = packet.Move().AsReadOnlySequence().Slice(5);
+
+        var signatureWriter = new ArrayBufferWriter<byte>();
+
+#if NET8_0
+        GetMICMethod(
```

Comment:
If this method isn't found, calling it will throw `MissingMethodException`.
We should add a catch for that in `TryAuthenticate`, and return `false`.

Reply:
Makes sense.